.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,31 @@
+name: codeql
+
+on:
+  push:
+    branches:
+    - master
+    - v1.9
+    - v1.8
+    - v1.7
+  pull_request:
+    branches:
+    - master
+  schedule:
+    - cron: "6 45 * * 3"
+
+jobs:
+  analyze:
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
```

Comment:
@twpayne Why is this necessary? It seems like each step in this job depends on the previous steps anyway, no?

Reply:
Is `fail-fast` only specific to a single GitHub workflow? From (possibly fallible) memory, at the time this PR was being developed, CI was very flaky, and CI flakes were causing the CodeQL analysis to be cancelled. I added this so that the analysis would run to completion.